sound/soc/sof/sof-client-probes.c

```diff
@@ -59,8 +60,8 @@ struct sof_ipc_probe_info_params {
 	struct sof_ipc_reply rhdr;
 	unsigned int num_elems;
 	union {
-		struct sof_probe_dma dma[0];
-		struct sof_probe_point_desc desc[0];
+		DECLARE_FLEX_ARRAY(struct sof_probe_dma, dma);
+		DECLARE_FLEX_ARRAY(struct sof_probe_point_desc, desc);
```

Comment:
that's the only change I was asking for, I don't get the long thread.

Reply:
Yep, I myself volunteered to fix all [0] issues at once, but hit a can of worms while doing it, and backed off.